src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -137,6 +137,105 @@ class Context:
     symbol_map: dict[str, IteratorExpr | ValueExpr | SymbolExpr]
 
 
+def _builtin_neighbors_indirect_addressing(
+    transformer: "PythonTaskletCodegen", node: itir.Expr, node_args: list[itir.Expr]
+) -> list[ValueExpr]:
+    sdfg: dace.SDFG = transformer.context.body
+    state: dace.SDFGState = transformer.context.state
+
+    offset_literal, data = node_args
+    assert isinstance(offset_literal, itir.OffsetLiteral)
+    iterator = transformer.visit(data)
+    offset_dim = offset_literal.value
+    assert isinstance(offset_dim, str)
+    table: NeighborTableOffsetProvider = transformer.offset_provider[offset_dim]
+    assert isinstance(table, NeighborTableOffsetProvider)
+
+    shifted_dim = table.origin_axis.value
+
+    result_name = unique_var_name()
+    sdfg.add_array(result_name, dtype=iterator.dtype, shape=(table.max_neighbors,), transient=True)
+    result_access = state.add_access(result_name)
+
+    table_name = connectivity_identifier(offset_dim)
+    table_array = sdfg.arrays[table_name]
+
+    me, mx = state.add_map(
+        f"{offset_dim}_neighbors_map",
+        ndrange={"neigh_idx": f"0:{table.max_neighbors}"},
+    )
+    shift_tasklet = state.add_tasklet(
+        "shift",
+        code="__result = __table[__idx, neigh_idx]",
+        inputs={"__table", "__idx"},
+        outputs={"__result"},
+    )
+    data_access_tasklet = state.add_tasklet(
```

Comment:
While running the icon4py diffusion tests, I encountered a problem on the ``data_access`` tasklet. The problem is that ``__field`` is supposed to have 1-d shape, but the memlet which is input to ``__field`` at line 202 is using the full shape. Validation does not catch this inconsistency, but the sdfg compilation fails.

Reply:
Fixed in [d04f60f](https://github.com/GridTools/gt4py/pull/1298/commits/d04f60ff954893a28edbc752bb2debd68e0b30e1)